`ShallowRef` should be exported

### Version
3.2.23

### Reproduction link
[github.com](https://github.com/sxzz/vue-types-issue)







### Steps to reproduce

```ts
import { defineComponent, shallowRef } from 'vue';

const App = defineComponent({
  setup() {
    const foo = shallowRef('bar');
    return {
      foo,
    };
  },
});

export { App };
```

- export `shallowRef` in setup
- generate dts
- it emits errors.

(There is no problem with `ref`.)

### What is expected?
emit no error.

### What is actually happening?
```
src/main.ts:3:7 - error TS4023: Exported variable 'App' has or is using name 'ShallowRefMarker' from external module "~/path/vue-types-issue/node_modules/.pnpm/@vue+reactivity@3.2.23/node_modules/@vue/reactivity/dist/reactivity" but cannot be named.

3 const App = defineComponent({
        ~~~


Found 1 error.
```

<!-- generated by vue-issues. DO NOT REMOVE -->

--- a/packages/reactivity/src/index.ts
+++ b/packages/reactivity/src/index.ts
@@ -12,6 +12,7 @@ export {
   ToRef,
   ToRefs,
   UnwrapRef,
+  ShallowRef,
   ShallowUnwrapRef,
   RefUnwrapBailTypes
 } from './ref'

--- a/packages/reactivity/src/ref.ts
+++ b/packages/reactivity/src/ref.ts
@@ -77,7 +77,7 @@ export function ref(value?: unknown) {
 
 declare const ShallowRefMarker: unique symbol
 
-type ShallowRef<T = any> = Ref<T> & { [ShallowRefMarker]?: true }
+export type ShallowRef<T = any> = Ref<T> & { [ShallowRefMarker]?: true }
 
 export function shallowRef<T extends object>(
   value: T

